src/main/kotlin/dev/mediamod/theme/impl/DynamicTheme.kt

```diff
@@ -15,99 +15,141 @@ class DynamicTheme : Theme.InbuiltTheme("Dynamic") {
-            } else {
-                getColors(quantizedColors)
-            }
-        } catch (e: Exception) {
-            e.printStackTrace()
-            return
-        }
-    }
 
-    private fun getContrastColors(quantizedColors: Array<Color>): Colors {
-        val background = quantizedColors[0]
-        val backgroundLuminance = background.luminance
+            // Use the most common color as the background
+            val background = quantizedColors[0].color
 
-        val progressBar = if (quantizedColors.size >= 2) {
-            quantizedColors[1].changeUntil(2f, background)
-        } else {
-            if (backgroundLuminance > 0.5) {
-                background.darker()
+            val progressBar = if (quantizedColors.size >= 2) {
+                // Of the rest of the colors with more than 10% of the pixels, find the one with the most contrast
+                // against the background
+                quantizedColors.subList(1, quantizedColors.size - 1)
+                    .filter { it.fraction > 0.10 }
+                    .maxByOrNull { it.color.contrast(background) }
+                    ?.color
+                    // If none of remaining colors have more than 10% of the pixels, use the 2nd most common color
+                    ?: quantizedColors[1].color
             } else {
-                background.brighter()
+                // Very rare case if image is all one color
+                background.changeBy(0.2f)
             }
-        }
 
-        val progressBarBackground = progressBar.changeUntil(1.3f)
-        val progressBarText = progressBar.changeUntil(3f)
+            // Create lighter/darker variants of the progress bar color for background and text
+            // The change parameter can be tuned to get different results
+            val progressBarBackground = progressBar.changeBy(0.15f)
+            val progressBarText = progressBar.changeBy(0.35f)
 
-        return Colors(
-            background = background,
-            text = Color.WHITE,
-            progressBar = progressBar,
-            progressBarBackground = progressBarBackground,
-            progressBarText = progressBarText
-        )
-    }
+            // Use black text if the background is bright
+            val text = if (background.luminance > 0.5) {
+                Color.BLACK
+            } else {
+                Color.WHITE
+            }
 
-    private fun getColors(quantizedColors: Array<Color>): Colors {
-        val progressBar = if (quantizedColors.size >= 2) {
-            quantizedColors[1]
-        } else {
-            quantizedColors[0].brighter()
+            colors = Colors(
+                background = background,
+                text = text,
+                progressBar = progressBar,
+                progressBarBackground = progressBarBackground,
+                progressBarText = progressBarText
+            )
+        } catch (e: Exception) {
+            e.printStackTrace()
```

Comment:
Use ``logger.error("An error occurred when doing color quantization: ", e)``

``logger`` is from ``dev.mediamod.utils``